Tasks/DotNetCoreInstallerV1/Tests/versionInstallerIsVersionInstalledTests.ts

```diff
@@ -0,0 +1,149 @@
+import * as tl from 'vsts-task-lib/task';
+import * as path from 'path';
+import { Constants } from "../versionutilities";
+let mockery = require('mockery');
+
+const version = "2.1.1";
+const installationPath: string = "installationPath"
+
+mockery.enable({
+    useCleanCache: true,
+    warnOnReplace: false,
+    warnOnUnregistered: false
+});
+
+mockery.registerMock('typed-rest-client/HttpClient', {
+    HttpClient: function () {
+        return {
+            get: function (url: string, headers) {
+                return "";
+            }
+        }
+    }
+});
+
+let sdkFolderExists = true;
+let sdkFileExists = true;
+let runtimeFolderExists = true;
+let runtimeFileExists = true;
+mockery.registerMock('vsts-task-lib/task', {
+    exist: function (elementPath: string) {
+        if (elementPath == path.join(installationPath, Constants.relativeSdkPath, version)) {
+            return sdkFolderExists;
+        }
+        else if (elementPath == path.join(installationPath, Constants.relativeSdkPath, `${version}.complete`)) {
+            return sdkFileExists;
+        }
+        else if (elementPath == path.join(installationPath, Constants.relativeRuntimePath, version)) {
+            return runtimeFolderExists;
+        }
+        else if (elementPath == path.join(installationPath, Constants.relativeRuntimePath, `${version}.complete`)) {
+            return runtimeFileExists;
+        }
+        else if (elementPath == installationPath) {
+            return true;
+        }
+
+        return false;
+    },
+    loc: function (locString, ...param: string[]) { return tl.loc(locString, param); },
+    debug: function (message) { return tl.debug(message); },
+    warning: function (message) { return tl.warning(message); },
+    error: function (errorMessage) { return tl.error(errorMessage); },
+    getVariable: function (variableName) { return tl.getVariable(variableName); },
+    getHttpProxyConfiguration: function () { return ""; },
+    setResourcePath: function (path) { return; }
+});
+
+import { VersionInstaller } from "../versioninstaller";
+let versionInstaller = new VersionInstaller("sdk", installationPath);
+
+if (process.env["__case__"] == "nonexplicit") {
+    let throwcount = 0;
+    try {
+        versionInstaller.isVersionInstalled("")
```

Comment:
Why can't this be in the caller and needs to be in this file?

Reply:
I thought this file only contains the mock

Reply:
This was the main L0 test file only focuses on validating the results using assertions and while this individual test files take care of invoking the function to be tested.
It helped in keeping a clean and maintainable L0 test file with code reuse via individual test files.